pkg/server/plugin/keymanager/repository.go

```diff
@@ -0,0 +1,17 @@
+package keymanager
+
+type Repository struct {
+	KeyManager KeyManager
+}
+
+func (repo *Repository) GetKeyManager() KeyManager {
```

Comment:
Does it make sense to try and standardize repository structure? We currently have singletons, lists, and maps. Wondering if it wouldn't be simpler for plugin developers (and potentially a little more future-proofed) if they were all maps?

Reply:
No, I don't think so. The original catalog interface was slices for all plugin types. This led to awkward code at each usage site that either:

1. Checked the length of the slice before using, returning an error if there was not the expected amount
2. In the case of singleton plugin types, just used element zero from the slice, e.g., `catalog.GetKeyManagers()[0]`

Similar to slices, maps would also pose similar problems. For example, if you wanted to get at the singleton key manager, you'd either have to know the name (yuck), or you'd have to do an awkward loop everywhere (also yuck), like:
```
var km keymanager.KeyManager
for _, plugin := range catalog.GetKeyManagers() {
     km = plugin
}
if km == nil {
    // ... die horribly because we don't have a keymanager that we definetely should have had
}
```

The server knows the "shape" or constraints of each plugin type. The catalog interface should be strongly aligned to those constraints. It makes the dependent code much clearer.

The pattern we've settled on is:

Things that may or may not be present, by name (i.e. node attestors and resolvers)
```
GetBlahBlahNamed(name string) (BlahBlah, bool)
```

Things that there are zero or more of (e.g. Notifiers):
```
GetBlahBlah() []BlahBlah
```

Things that are are exactly one of (e.g. KeyManager):
```
GetBlahBlah() BlahBlah
```

Things that there may be one of (e.g. UpstreamAuthority)
```
GetBlahBlah() (BlahBlah, bool)
```

Reply:
Hmm. I definitely see the issue with using list - no names or other identifying features. For singletons (or at least current singletons) a standard default name like "default" should suffice. I was considering theoretical future cases where multiple key managers would be supported. Converting from a singleton to a map at that point would necessitate updating all key managers simultaneously. Similarly, for Notifiers which is currently a list, using a map doesn't meaningfully impact the interaction.

All that said, I do appreciate crossing those respective bridges if and when we need to - YAGNI (Martin Fowler)

Reply:
If we ever move from singleton to slice or map, the shape change of the catalog is going to be the least of our worries.

Yes, YAGNI :)